ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java

```diff
@@ -257,39 +260,59 @@ String format(final Bytes bytes) {
 
   enum WindowSchema {
 
-    SESSION(SessionWindowedDeserializer::new),
+    SESSION(WindowSchema::newSessionWindowedDeserializer),
     HOPPING(WindowSchema::newTimeWindowedDeserializer),
     TUMBLING(WindowSchema::newTimeWindowedDeserializer);
 
-    private final Function<Deserializer<?>, Deserializer<?>> mapper;
+    private final Function<NamedDeserializer, Deserializer<?>> mapper;
 
-    WindowSchema(final Function<Deserializer<?>, Deserializer<?>> mapper) {
+    WindowSchema(final Function<NamedDeserializer, Deserializer<?>> mapper) {
       this.mapper = requireNonNull(mapper, "mapper");
     }
 
     public NamedDeserializer wrap(final NamedDeserializer inner) {
 
       final String name = name() + "(" + inner.name + ")";
 
-      final Deserializer<?> deserializer = mapper.apply(inner.deserializer);
+      final Deserializer<?> deserializer = mapper.apply(inner);
 
-      return new NamedDeserializer(name, inner.doNotWrap, deserializer);
+      return new NamedDeserializer(name, inner.doNotWrap, deserializer, inner::format);
     }
 
-    private static Deserializer<?> newTimeWindowedDeserializer(final Deserializer<?> inner) {
-      final TimeWindowedDeserializer<?> windowedDeser = new TimeWindowedDeserializer<>(inner);
+    private static Deserializer<?> newSessionWindowedDeserializer(
+        final NamedDeserializer inner
+    ) {
+      final SessionWindowedDeserializer<?> sessionDeser
+          = new SessionWindowedDeserializer<>(inner.deserializer);
+
+      return (topic, data) -> {
+        final Windowed<?> windowed = sessionDeser.deserialize(topic, data);
+        return "[" + inner.format(windowed.key())
+            + "@" + windowed.window().start() + "/" + windowed.window().end() + "]";
+      };
+    }
+
+    private static Deserializer<?> newTimeWindowedDeserializer(
+        final NamedDeserializer inner
+    ) {
+      final TimeWindowedDeserializer<?> windowedDeser
+          = new TimeWindowedDeserializer<>(inner.deserializer);
 
       return (topic, data) -> {
         final Windowed<?> windowed = windowedDeser.deserialize(topic, data);
 
         // Exclude window end time for time-windowed as the end time is not in the serialized data:
-        return "[" + windowed.key() + "@" + windowed.window().start() + "/-]";
+        return "[" + inner.format(windowed.key()) + "@" + windowed.window().start() + "/-]";
       };
     }
   }
 
   enum Format {
-    AVRO(0, KafkaAvroDeserializer::new),
+    AVRO(0, KafkaAvroDeserializer::new, Object::toString),
+    PROTOBUF(
+        0,
+        KafkaProtobufDeserializer::new,
+            o -> TextFormat.printer().shortDebugString((Message) o)),
```

Comment:
Off-topic, but just wanted to check you have it on your radar to split out a `SR_JSON` out of the `JSON` format?

Reply:
Found it: https://github.com/confluentinc/ksql/pull/4596